### ISSUE TITLE
Passport gets stuck on animate-out after saving

### ISSUE THREAD

**Issue by Richard-L** [LostArtefacts/TRX#1335]
[TR1X.4.0.3-28-g211d744-Windows.zip](https://github.com/LostArtefacts/TR1X/releases/download/latest/TR1X.4.0.3-28-g211d744-Windows.zip) on the left and last release on the right:

https://github.com/LostArtefacts/TR1X/assets/4934209/3c925ca6-fcc2-4744-a37b-ff446e905fa4

Notice how on the left the passport gets stuck for a few frames before plopping away at the end of the animation.

**Comment by rr-** [comment]
I can't quite reproduce it – can someone confirm? @LostArtefacts/dev

**Comment by lahm86** [comment]
I'm struggling to see it, perhaps it can be re-tested after #1339 is merged (it looks good to me in there currently).

**Comment by rr-** [comment]
Merged #1339

**Comment by walkawayy** [comment]
> Merged #1339

Yes I just tested on -2 speed, and I don't see an issue. I think it was a side effect of the jitter bug.

**Comment by rr-** [comment]
I think it may have to do with the phaser refactor, where executing the savegame routines under lags may freeze inventory at the last frame, whereas it could have been a frame later (where the inventory spinout is done animating).

**Comment by rr-** [comment]
@Richard-L can you check if the issue persists in 4.1? We stumbled across this very bug when fixing fade speeds for 60 FPS+Vsync+Turbo cheat and have since fixed it, maybe it fixed your scenario as well.

**Comment by Richard-L** [comment]
@rr- It's definitely better. Still want to think it hangs for a few milliseconds. But at this point I'm unsure if I would've noticed it in this state, and how objective I can still be.
Perhaps could you point me to one of the first 60fps builds? Could do a frame by frame analysis if we wanted.

**Comment by rr-** [comment]
>Perhaps could you point me to one of the first 60fps builds? Could do a frame by frame analysis if we wanted.

I don't have these anymore, and we've linearized git history which means they're inaccessible.

**Comment by rr-** [comment]
Can someone confirm when the saving to disk occurs – if it's at the frame where the lag occurs?
(It's easy to highlight with an artificial SDL sleep function call.)
